nuxt/index.js

```diff
@@ -1,39 +1,64 @@
 const { resolve } = require('path')
 
-function pickFirst(...args) {
-  for (const arg of args) {
-    if (arg !== undefined) {
-      return arg
+module.exports = function nuxtBootstrapVue(moduleOptions = {}) {
+  this.nuxt.hook('build:before', () => {
+    function pickFirst(...args) {
+      for (const arg of args) {
+        if (arg !== undefined) {
+          return arg
+        }
+      }
     }
-  }
-}
 
-module.exports = function nuxtBootstrapVue(moduleOptions = {}) {
-  // Merge moduleOptions with default
-  const options = {
-    ...this.options.bootstrapVue,
-    ...moduleOptions
-  }
-
-  const bootstrapVueCSS = pickFirst(options.bootstrapVueCSS, options.bootstrapVueCss, options.bvCSS)
-  if (bootstrapVueCSS) {
-    // Add BootstrapVue CSS
-    this.options.css.unshift('bootstrap-vue/dist/bootstrap-vue.css')
-  }
-
-  const bootstrapCSS = pickFirst(options.bootstrapCSS, options.bootstrapCss, options.css)
-  if (bootstrapCSS) {
-    // Add Bootstrap CSS
-    this.options.css.unshift('bootstrap/dist/css/bootstrap.css')
-  }
-
-  // Transpile src
-  this.options.build.transpile.push('bootstrap-vue/src')
-
-  // Register plugin, pasing options to plugin template
-  this.addPlugin({
-    src: resolve(__dirname, 'plugin.template.js'),
-    fileName: 'bootstrap-vue.js'
+    const kebabCase = str => str.replace(/([_\s]+([a-zA-Z])|([A-Z]))/g, (m, $1, $2, $3, o) => (o ? '-' : '') + ($2 || $3 || '').toLowerCase())
+    const pascalCase = str => str.replace(/(^|[-_\s]+)(.)/g, (m, $1, $2) => $2.toUpperCase())
```

Comment:
I included these here because I think a oneliner is preferable above an external dependency

Reply:
@pi0 @tmorehouse: Shouldn't we use the `lodash` utils here or is this JS going into a browser bundle?

Reply:
It is only ever run server side during nuxt build

Reply:
Using lodash is against the efforts that @pi0 put in #2523 

lodash has a lot of extra code with regards to unicode chars which you dont need here, as we know that plugin names are ascii chars only so lodash seemed not the correct choice to me. Also there didnt seem to be other use cases in bootstrap-vue for these conversions so thats why I choice not to create more dependency trees and keep the plugin as flat as possible as the nuxt plugin is already something of an oddball in this repo :)